include/IO/mesh/fortran/read_boundaries.hpp

```diff
@@ -0,0 +1,75 @@
+#pragma once
+
+#include "mesh/boundaries/absorbing_boundaries.hpp"
+#include "mesh/boundaries/acoustic_free_surface.hpp"
+#include "mesh/boundaries/boundaries.hpp"
+#include "mesh/boundaries/forcing_boundaries.hpp"
+#include "specfem_mpi/interface.hpp"
+#include <fstream>
+#include <vector>
+
+namespace specfem {
+namespace IO {
+namespace mesh {
```

Comment:
Can we move these functions within the `impl` namespace i.e. `specfem::IO::mesh::impl::fortran::read_boundaries`. Functions within `impl` namespace are implementation details and are not meant to exposed to the developer. I started following this convension recently so only the recent implementations are compliant with this philosophy.

Reply:
doing it as part of #185 .